helix-term/src/ui/text_decorations.rs

```diff
@@ -144,6 +153,37 @@ impl<'a> DecorationManager<'a> {
     }
 }
 
+pub struct ColorSwatch {
```

Comment:
looking at this closer, this should act more like a state machine and we should only have one per document see this commit for how it can be implemented this way https://github.com/gabydd/helix/commit/3bfcbb555169c1c8b2c668abd70aa9f8acd198ee

Reply:
Thanks!